client/layout/guided-tours/docs/API.md

```diff
@@ -0,0 +1,162 @@
+# Guided Tours API
+
+Guided Tours are declared as a tree of JSX components.
+
+
+## Tour
+
+Tour is a React component that declares the top-level of a tour. It defines conditions for starting a tour and contains `<Step>` components as children.
+
+### Props
+
+* `name`: (string) Unique name of tour in camelCase.
+* `version` (string): Version identifier. We use date string like "20161224".
+* `path` (string or array, optional): Use this prop to limit tour only to some path prefix (or more prefixes if array). Example: `[ '/stats', '/settings' ]`
+* `when` (function, optional): This is a redux selector function. Use this to define conditions for the tour to start.
+
+## Step
+
+Step is a React component that defines a single Step of a tour. It is represented as dark box on top of Calypso UI. Step can be positioned in many ways in relation to any DOM node in Calypso that is marked with `[data-tip-target]` attribute.
+
+### Props
+
+* `name`: (string) Unique identifier of the step.
+* `target`: (string, optional) Target which this step belongs to and will be used for positioning. This prop is value is used to look up according `[data-tip-target]` in DOM. If omitted, `<body>` will be used as target. TODO: mention querySelector hack?, is body really default target?
```

Comment:
> querySelector hack

Even though it was intended as a secondary mechanism, we do use it a bit and has a few interesting strengths, like in when `.some-container .some-state-like-is-valid .my-target`. We should mention it.

> `<body>` will be used

Well, in a way. `positioning` will grab `body.getBoundingClientRect()` in a [few places](https://github.com/Automattic/wp-calypso/blob/8d290ccfed6d3219912df71b159a631285b7590c/client/layout/guided-tours/positioning.js#L103) if no target is found, but I wouldn't say that `body` is a "default target". Failing to provide a `target` will mess up positioning with most `placement` methods, unless they're meant for that.